### DISCUSSION TITLE
"Percentage of total" metric

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#365] [Anything Else]
"t would be nice to have a % based view. Currently all the numbers are absolute. It would be nice to know how much % of users got converted. I currently do the mental math in my head, by dividing by the number of unique visits. The same way for "Top Sources", I have concrete numbers, but I would like to know how many % of users came from which site. The same thing for "Top Pages" - with percentages, I can easily see "Ah. 10% of the users visit the support page, and 50% of them go to the pricing page". I could achieve this by making each page a goal, but that's cumbersome."

[request via email] 

With all this filtering we now allow, could be interesting to add a "Percentage of total" metric to the top chart (where we have unique visitors, total pageviews, bounce rate, visit duration). So whatever you click and filter the dashboard by, there on the top you can easily see say Hacker News sent 720 unique visitors which is 25% of your total traffic in this time period. Same for pages, countries, devices, goal conversions and everything else.

**Comment by Vigasaurus** [comment]
How would we feel about this combining with #117? i.e. a user picks one of the four metrics at the top to see in graph form (I'm unsure if we actually store the hour by hour data for the last two, I'd have check) and then the same is used as the denominator for these calculations

**Reply by metmarkosaric** [reply]
Could be great combined for sure @Vigasaurus! Having "percentage of total" in addition to being able to click on a metric and switch the main graph to show that metric would make Plausible so much more powerful.

**Reply by ukutaht** [reply]
I'm not sure I understand how these two features combine. If I click on bounce rate, what metric would be shown in the Top Pages report for example? According to the logic it would be `pageviews / total_bounce_rate` which feels like would be a nonsense metric?

It does make sense for when you click on `pageviews` or `visitors` but I can't quite imagine how it would work with bounce rate and session length

**Reply by Vigasaurus** [reply]
I imagined that it was less so a combination and more so just that the same user input for selection would apply to both - so it'd still only use views or users as the denominator, but the graph would change with the same clicks but for all 4. 

It'd definitely be a bit of a big thing to make sure that the edge cases for the percentages get handled - and I don't think it'd actually end up being done by me since I've realized my elixir is too weak for it heh.

**Reply by ukutaht** [reply]
Ah OK I see. That sounds good as a feature. If you want you can sketch it out on the frontend with dummy data and I can hook it up to the backend.

If you're looking to learn Elixir, @felipesere, @iamvery and myself maintain [Elixir koans](https://github.com/elixirkoans/elixir-koans) that will get you up to speed.

**Reply by Vigasaurus** [reply]
Sounds good! I may look into doing something similar for various other medium-ish features - i.e. scaffolding the front-end out and letting someone with stronger elixir top up the backend - since frontend is generally my strong suit anyways (apart from Node backends). 
> Elixir koans that will get you up to speed

Cool - I'll look into that

**Comment by tcurdt** [comment]
Glad I am not alone on this. I found this discussion because the goals are only absolute.

It feels really weird I have the absolute `Uniques` and `Total conversions` for the goals, but having to do the ratio of `UNIQUE VISITORS` manually. And there is soooo much space in the `Goals Conversion` table.

**Reply by ukutaht** [reply]
@tcurdt Thanks for the feedback, I'd like to ask some clarification to make sure I understand completely.

Conversion rate is shown at the moment for each goal. When you click on a goal, we also show a conversion rate at the top. What ratio do you have to do manually at the moment that we don't show already?

**Reply by tcurdt** [reply]
If the percentage shown on the goal page would make it into the overview table (without clicking on the individual goal) that would be great.

**Reply by metmarkosaric** [reply]
the conversion rate is already shown on the overview table for quick overview. check our live demo https://plausible.io/plausible.io

**Reply by tcurdt** [reply]
Ah, OK. Exactly what I was hoping for :) I guess that's a difference to the self-hosted version then.

**Reply by metmarkosaric** [reply]
ahh i see. you're on the self-hosted version. self-hosted is not updated as frequently as the hosted version which changes daily. 

think of it as a long term release to help us and help you not need to upgrade things all the time and risk breaking stuff. we stack all the new features and fixes that have been battle tested on the hosted version and release a new self-hosted version twice a year. you can learn more here: https://plausible.io/blog/building-open-source

version 1.2 of self-hosting is ready to be beta tested and this one has the feature you're looking for. see https://github.com/plausible/analytics/discussions/605

and if you're enjoying self-hosted Plausible and finding it useful, please do consider sponsoring the project to give back to the community and to contribute to the long-term sustainability of the project. see details here: https://github.com/sponsors/plausible

**Reply by tcurdt** [reply]
Thanks. I am aware of the LTR approach - but didn't think to cross-check. Doh! Sorry for the noise :)

**Reply by metmarkosaric** [reply]
no problem!

**Comment by Keavon** [comment]
This is really basic and so important for making conclusions about trends in the data, especially as the total number of site visitors grows over the years. Please implement this feature.

**Comment by LushawnDev** [comment]
Is there any update on this? I'm manually having to calculate percentages for things like page views at the moment. This functionality, even as a simple switch between absolute or percentage would be such a big improvement!
